webpack/scenes/ContentViews/ContentViewsCountHeader.js

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Spinner } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { STATUS } from 'foremanReact/constants';
+import ContentViewIcon from './components/ContentViewIcon';
+
+const ContentViewsCountHeader = ({ response, status }) => {
+  const [componentCount, setComponentCount] = useState(<Spinner size="sm" />);
+  const [compositeCount, setCompositeCount] = useState(<Spinner size="sm" />);
+  const { results } = response;
+  const loadingResponse = status === STATUS.PENDING;
+
+  useEffect(
+    () => {
+      if (loadingResponse) {
+        setComponentCount(<Spinner size="sm" />);
+        setCompositeCount(<Spinner size="sm" />);
+      }
+      if (!loadingResponse && results) {
+        let componentCVCount = 0;
+        let compositeCVCount = 0;
+        results.forEach((contentView) => {
+          if (contentView.composite) {
+            compositeCVCount += 1;
+          } else {
+            componentCVCount += 1;
+          }
+        });
+        setComponentCount(componentCVCount);
+        setCompositeCount(compositeCVCount);
+      }
+    },
+    [JSON.stringify(response), status],
```

Comment:
I still hate doing `JSON.stringify` in the dependency array.  Would you consider adding https://github.com/kentcdodds/use-deep-compare-effect ?

Reply:
In fact, I don't think this `useEffect` is needed at all:

1. see comment above about `loading` - can pass the value of `loadingResponse` directly to the child component
2. For results, you can just use `filter`:
```js
const compositeCVCount = results.filter(contentView => !!contentView.composite).length
const componentCVCount = results.filter(contentView => !contentView.composite).length
```

Come to think of it, this completely eliminates the need to store the counts with `useState` as well.

(If this causes too many renders or performance issues, could possibly wrap it in a `useMemo`.)